File: U3EntrenamientoApp/app/src/main/java/mx/edu/itl/u3entrenamientoapp/CustomListAdapter.java
/*------------------------------------------------------------------------------------------
:*                         TECNOLOGICO NACIONAL DE MEXICO
:*                                CAMPUS LA LAGUNA
:*                     INGENIERIA EN SISTEMAS COMPUTACIONALES
:*                             DESARROLLO EN ANDROID "A"
:*
:*                   SEMESTRE: AGO-DIC/2023    HORA: 08-09 HRS
:*
:*                       Clase del adaptador personalizado
:*
:*  Archivo     : CustomListAdapterActivity.java
:*  Autor       : Jorge Cisneros de la Torre     20130789
:*  Fecha       : 26/Oct/2023
:*  Compilador  : Android Studio Giraffe 2022.3.1
:*  Descripción : Clase del adaptador personalizado que hace uso de la clase de
                  ItemModel para permitir que se vean las imagenes y el texto
                  con el diseño que se creo (item_lista.xml)
:*  Ultima modif:
:*  Fecha       Modificó             Motivo
:*==========================================================================================
:*
:*------------------------------------------------------------------------------------------*/

package mx.edu.itl.u3entrenamientoapp;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

//--------------------------------------------------------------------------------------------------

public class CustomListAdapter extends BaseAdapter  {

    Context context;
    List<ItemModel> items;
    LayoutInflater inflador;

    //----------------------------------------------------------------------------------------------

    public CustomListAdapter ( Context contexto, List<ItemModel> elementos ) {
        this.context = contexto;
        this.items = elementos;
        // inflamos con el contexto que se pase al constructor
        inflador = LayoutInflater.from( contexto );
    }

    //----------------------------------------------------------------------------------------------

    // Tamaño de la lista
    @Override
    public int getCount() {
        return items.size ();
    }

    //----------------------------------------------------------------------------------------------

    @Override
    public Object getItem( int position ) {
        return items.get( position );
    }

    //----------------------------------------------------------------------------------------------

    @Override
    public long getItemId( int position ) {
        return position;
    }

    //----------------------------------------------------------------------------------------------

    @Override
    public View getView( int position, View convertView, ViewGroup parent ) {
        View view = convertView;

        // Obtener el diseño de los elementos de la lista
        if ( view == null ) {
            view = LayoutInflater.from( context ).inflate( R.layout.item_lista, parent, false );
        }

        // Obtener las imagenes y texto
        ImageView imageView = view.findViewById( R.id.imagen );
        TextView textView = view.findViewById( R.id.texto );

        ItemModel item = items.get( position );

        // Colocar la imagen y texto específico para cada elemento de la lista
        imageView.setImageResource( item.getRecursoImg() );
        textView.setText( item.getTexto() );

        return view;
    }

    //----------------------------------------------------------------------------------------------

}

File: PolarisApp/app/src/main/java/mx/edu/itl/polarisapp/lista/CustomListAdapter.kt
/***************************************************************************************************
                CustomAdapter.kt Última modificación: 21/Noviembre/2023
***************************************************************************************************/
package mx.edu.itl.polarisapp.lista

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import androidx.compose.foundation.shape.RoundedCornerShape
import com.bumptech.glide.Glide
import mx.edu.itl.polarisapp.EventosActivity
import mx.edu.itl.polarisapp.R

class CustomListAdapter ( contexto : Context, elementos : List<ItemModel> ) : BaseAdapter () {

    var context : Context = contexto
    var items : List<ItemModel> = elementos

     var inflater : LayoutInflater

    init {
        inflater = LayoutInflater.from ( context )
    }
    //----------------------------------------------------------------------------------------------
    override fun getCount (): Int {
        return items.size
    }
    //----------------------------------------------------------------------------------------------
    override fun getItem ( position: Int ) : Any {
        return items.get ( position )
    }
    //----------------------------------------------------------------------------------------------
    override fun getItemId ( position: Int ) : Long {
        return position.toLong ()
    }
    //----------------------------------------------------------------------------------------------
    override fun getView ( position : Int, convertView : View?, parent : ViewGroup ) : View {
        var view = convertView

        if ( view == null ) {
            view = inflater.inflate ( R.layout.elementos_eventos, parent, false )
        }

        // se asignan los widgets a las variables para poderles asignar el contenido
        var imagenEvento : ImageView = view!!.findViewById ( R.id.imgEvento )
        var txtTitulo : TextView = view.findViewById ( R.id.txtTitulo )
        var txtFecha : TextView = view.findViewById ( R.id.txtFecha )
        var txtHorario : TextView = view.findViewById ( R.id.txtHoraInicFin )
        var txtLugar : TextView = view.findViewById ( R.id.txtLugar )
        var txtDescripcion : TextView = view.findViewById ( R.id.txtDescripcion )

        imagenEvento.setImageResource ( R.drawable.eventosdefault )

        txtTitulo.setText ( items.get ( position ).getTitulo () )
        txtFecha.setText ( items.get ( position ).getFecha () )
        txtHorario.setText ( items.get ( position ).getHoraA () + "-"+items.get ( position ).getHoraC () )
        txtLugar.setText ( items.get ( position ).getLugar () )
        txtDescripcion.setText ( items. get ( position ).getDescripcion () )

        return view
    }
}